### ISSUE TITLE
Loading a custom metric renames it to 'python_function'

### ISSUE THREAD

**Issue by OliverHofkens** [rstudio/keras#184]
If I train a model with a custom metric in it, then save the model with `save_model_hdf5`, then load it again with `load_model_hdf5(custom_objects=c(custom_metric_name=custom_metric))` the metric is then known as 'python_function'.
Feel free to correct me if I'm doing something wrong, the docs are a bit unclear about loading custom objects. 

Reproducible example:
Use following metric in a model:
```
sparse_top_k_cat_acc <- function(y_pred, y_true){
    metric_sparse_top_k_categorical_accuracy(y_pred, y_true, k = FLAGS$topKMetric)
}
...
model %>% compile(
         loss = "sparse_categorical_crossentropy",
         optimizer = optimizer_nadam(lr = FLAGS$learningRate),
         metrics = c(top_k_acc = sparse_top_k_cat_acc)
)
```

Then save the model:
```
save_model_hdf5(model, 'test.hdf5')
```

Then load the model again:
```
model <- load_model_hdf5('test.hdf5', custom_objects=c(top_k_acc=sparse_top_k_cat_acc))
```

When training, the following message appears:

> RuntimeWarning: Early stopping conditioned on metric `top_k_acc` which is not available. Available metrics are: loss,**python_function**,lr,val_loss,**val_python_function**

**Comment by jjallaire** [comment]
If you do this when defining your custom metric then it should work:

```r
sparse_top_k_cat_acc <- function(y_pred, y_true){
  metric_sparse_top_k_categorical_accuracy(y_pred, y_true, k = 4)
}
attr(sparse_top_k_cat_acc, "py_function_name") <- "sparse_top_k_cat_acc"
```
 
I have a TODO item to make this cleaner/easier.

**Comment by OliverHofkens** [comment]
Great, thank you!

**Comment by jjallaire** [comment]
This should now work correctly without requiring you to do the `attr()` business. e.g. see https://github.com/rstudio/keras/blob/master/tests/testthat/test-model-persistence.R#L17

Could you let know if this does indeed work correctly sans the previously described workaround in your use case?

**Comment by jjallaire** [comment]
There's also the just added `with_custom_object_scope()` function: https://keras.rstudio.com/reference/with_custom_object_scope.html
